public/app/vip-boot.js

```
@@ -211,11 +211,44 @@
+
+    function updateNetPill(online) {
+      setEnginePill('engNetPill', online ? 'ready' : 'error');
+    }
+
+    // Set initial state immediately
+    var isOnline = (typeof navigator !== 'undefined' && typeof navigator.onLine === 'boolean')
+      ? navigator.onLine
+      : true;
+    updateNetPill(isOnline);
+
+    window.addEventListener('online', function () {
```

ESLint

[error] 'window' is not defined.

(no-undef)